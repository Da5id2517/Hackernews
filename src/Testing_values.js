export const list = [
  {
    title: 'Naslov',
    url: 'https://lemonparty.org',
    author: 'SomeBloke',
    num_comment: 3,
    points: 4,
    objectID: 0
  },
  {
    title: 'Seheron',
    url: 'https://youtube.org',
    author: 'Tseries',
    num_comment: 12,
    points: 3,
    objectID: 1
  }
]

export const blank_func = () => {};
export const test_string = 'test_string';
