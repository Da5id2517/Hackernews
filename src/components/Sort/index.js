import React from 'react';
import classNames from 'classnames';
import Button from '../Button';
import PropTypes from 'prop-types';


const Sort = ({sortKey, onSort, activeSortKey, children}) => {
  const sortClass = classNames(
    'button-inline',
    { 'button-active': sortKey === activeSortKey}
  );

  return(
    <Button
      onClick={() => onSort(sortKey)}
      className={sortClass}>
      {children}
    </Button>
  );
}

Sort.propTypes = {
  sortKey: PropTypes.string.isRequired,
  onSort: PropTypes.func.isRequired,
  activeSortKey: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired
};


export default Sort;
