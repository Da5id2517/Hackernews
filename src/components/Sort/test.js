import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Sort from './';
import {test_string, blank_func} from '../../Testing_values.js';


describe('Search', () => {
  it('renders properly', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <Sort
      onSort={blank_func}
      activeSortKey={test_string}
      sortKey={test_string}>
        {test_string}
      </Sort>
      , div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('has valid snap', () => {
    const comp = renderer.create(
      <Sort
      onSort={blank_func}
      activeSortKey={test_string}
      sortKey={test_string}>
        {test_string}
      </Sort>
    );
    let tree = comp.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
