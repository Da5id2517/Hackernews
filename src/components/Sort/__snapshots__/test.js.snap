// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search has valid snap 1`] = `
<button
  className="button-inline button-active"
  onClick={[Function]}
  type="button"
>
  test_string
</button>
`;
